components/clamav/files/freshclam.xml

```diff
@@ -0,0 +1,36 @@
+<?xml version='1.0'?>
+<!DOCTYPE service_bundle SYSTEM '/usr/share/lib/xml/dtd/service_bundle.dtd.1'>
+<service_bundle type='manifest' name='clamav'>
+  <service name='antivirus/freshclam' type='service' version='0'>
+    <single_instance/>
+    <dependency name='fs' grouping='require_all' restart_on='none' type='service'>
+      <service_fmri value='svc:/system/filesystem/local'/>
+    </dependency>
+    <dependency name='network' grouping='require_all' restart_on='none' type='service'>
+      <service_fmri value='svc:/milestone/multi-user-server'/>
+    </dependency>
+    <instance name='default' enabled='false'>
+      <exec_method type='method'
+                   name='start'
+                   exec='while true ; do /usr/bin/freshclam.sh &amp;&amp; echo OK || echo FAILED ; sleep 900; echo Refreshing ClamAV database; done'
```

Comment:
Hopefully made better by a recent commit. Review and testing is advised ;)
Now that this is not a loop, nothing forbids this from queuing up in case of some long lags or slow downloads; however the cron schedule (enabled upon every service startup, removed upon every shutdown) should be configurable as a variable in the SMF service so particular installations might have a way to amend this ;)